auth/server/src/main/java/com/seagates3/policy/PolicyValidator.java

```diff
@@ -229,4 +238,170 @@ boolean isPrincipalIdValid(Principal principal) {
       validateActionAndResource(List<Action> actionList,
                                 List<Resource> resourceValues,
                                 String inputBucket);
-}
+
+ protected
+  ServerResponse validateVersion(String versionValue) {
+    ServerResponse response = null;
+    String policyVersion = AuthServerConfig.getPolicyVersion();
+    if (versionValue != null) {
+      if (versionValue.isEmpty()) {
+        response =
+            responseGenerator.malformedPolicy("Syntax errors in policy.");
+        LOGGER.error("Version field cannot be empty");
+        return response;
+
+      } else if (!versionValue.equals(policyVersion)) {
+        response = responseGenerator.malformedPolicy(
+            "Policy document must have version " + policyVersion +
+            " or greater.");
+        LOGGER.error("Policy document must have version " + policyVersion +
+                     " or greater.");
+        return response;
+      }
+    }
+    return response;
+  }
+ protected
+  ServerResponse validateStatementSyntax(JSONObject jsonObject) {
+    ServerResponse response = null;
+    if (jsonObject.get(JsonDocumentFields.STATEMENT) instanceof JSONArray) {
+      JSONArray arr = (JSONArray)jsonObject.get(JsonDocumentFields.STATEMENT);
+      if (arr.length() == 0) {
+        response =
+            responseGenerator.malformedPolicy("Syntax errors in policy.");
+        LOGGER.error("Statement array can not be empty");
```

Comment:
print policy name

Reply:
Policy name is not available here. 
But I have added the debug log with policy name included at the time of calling validation from PolicyController#Create.